bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Button.java

```diff
@@ -299,7 +299,22 @@ int computeLeftMargin () {
 		if (newFont != 0) OS.SelectObject (hDC, oldFont);
 		OS.ReleaseDC (handle, hDC);
 		OS.GetClientRect (handle, rect);
-		margin = Math.max (MARGIN, (rect.right - rect.left - margin) / 2);
+		//in case of left alignment
+		if ((style & SWT.LEFT) != 0) {
+			margin = MARGIN;
+		}
+		//in case of center alignmnet
+		else if ((style & SWT.CENTER) != 0) {
+			margin = Math.max (MARGIN, (rect.right - rect.left - margin) / 2);
+		}
+		//in case of right alignment
+		else if ((style & SWT.RIGHT) != 0) {
+			margin = Math.max (MARGIN, (rect.right - rect.left - margin - MARGIN));
+		}
+		//no alignment which is defaulted to center.
+		else {
+			margin = Math.max (MARGIN, (rect.right - rect.left - margin) / 2);
+		}
```

Comment:
- These comments are not useful should be removed.
- Also if-else can be simplified as below:
if ((style & SWT.LEFT) != 0) {
	margin = MARGIN;
} else if ((style & SWT.RIGHT) != 0) {
	margin = Math.max (MARGIN, (rect.right - rect.left - margin - MARGIN));
} else {
	margin = Math.max (MARGIN, (rect.right - rect.left - margin) / 2);
}